index.html

```diff
@@ -3,47 +3,279 @@
+        if (options.filtered === 'region-ne') {
+           data = data.filter(function (c) {
+              return c.Region === "Northeast"; });
+        } 
+    }    
+
+    // Select only the schools from the current page (slice)
+
+
+
+
+    var table = document.getElementById('data-table');
+    var tbody = table.getElementsByTagName('tbody')[0];
+    
+    tbody.innerHTML = '';
+
+    data.forEach(function (d) {
+      var row = document.createElement('tr');
+      
+      var vote = document.createElement('td');
+      vote.innerHTML = d.State;
+      row.appendChild(vote);
+
+      var region = document.createElement('td');
+      region.innerHTML = d.Region;
+      row.appendChild(region);
+
+      var repVotes = document.createElement('td');
+      repVotes.innerHTML = d.RepVotes;
+      row.appendChild(repVotes);
+
+      var totalVotes = document.createElement('td');
+      totalVotes.innerHTML = d.TotalVotes;
+      row.appendChild(totalVotes);
+
+      var repPercentage = document.createElement('td');
+      repPercentage.innerHTML = d.Percentage;
+      row.appendChild(repPercentage);
+
+      var barWrapper = document.createElement('td');
+      var bar = document.createElement('div');
+      bar.setAttribute('class', 'bar');
+      
+      var repPerc = d.perc / MAX_PERCENT;
+      bar.setAttribute('style', 'width: ' + (repPerc * 100) + '%;');
+      barWrapper.appendChild(bar);
+      row.appendChild(barWrapper);
+
+      tbody.appendChild(row);
+
+      });
+  }
+
+  updateTable();
+
+  document.getElementById('sort-asc').addEventListener('click', function() {
+    if (options.sorted === 'asc') {
+      options.sorted = false;
+    } else {
+      options.sorted = 'asc';
+    }
+    updateTable();
+  });
+
+  document.getElementById('sort-desc').addEventListener('click', function () {
+    if (options.sorted === 'desc') {
+      options.sorted = false;
+    } else {
+      options.sorted = 'desc';
+    }
+    updateTable();
+  });
+
+  document.getElementById('filter-w').addEventListener('click', function () {
+    if (options.filtered === 'region-w') {
+      options.filtered = false;
+    } else {
+      options.filtered = 'region-w';
+    }
+    updateTable();
+  });
+
+  document.getElementById('filter-mw').addEventListener('click', function () {
+    if (options.filtered === 'region-mw') {
+      options.filtered = false;
+    } else {
+      options.filtered = 'region-mw';
+    }
+    updateTable();
+  });
+
+  document.getElementById('filter-s').addEventListener('click', function () {
+    if (options.filtered === 'region-s') {
+      options.filtered = false;
+    } else {
+      options.filtered = 'region-s';
+    }
+    updateTable();
+  });
+
+  document.getElementById('filter-ne').addEventListener('click', function () {
+    if (options.filtered === 'region-ne') {
+      options.filtered = false;
+    } else {
+      options.filtered = 'region-ne';
+    }
+    updateTable();
+  });
+
```

Comment:
Yup, this works great. For next week, try adding in a selected state (Bootstrap gives that to us if we add the class `active` to a button) using `d3.selectAll(...).classed('active', true/false)`.